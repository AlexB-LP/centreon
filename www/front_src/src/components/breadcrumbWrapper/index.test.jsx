import React from 'react';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter, Router, Route, Switch } from 'react-router-dom';
import BreadcrumbWrapper from '.';
//import { ConnectedRouter } from "react-router-redux";
//import { createBrowserHistory } from "history";
import { createMemoryHistory } from 'history'

//import MockRouter from 'react-mock-router';

/*
import { Link } from 'react-router-dom';

Link = jest.fn();
*/

//jest.mock('react-router-dom/Link', () => 'a')

/*
jest.mock('react-router-dom/Link', () => () => 'link');
jest.mock('react-router-dom', () => ({
  __esModule: true, // this property makes it work
  default: 'mockedDefaultExport',
  Link: jest.fn(),
}));
*/

//jest.mock('react-router-dom/Link', () => () => 'a')

const mockStore = configureStore([]);
const history = createMemoryHistory({});
//const history = createBrowserHistory({basename: '/'});

describe('BreadcrumbWrapper', () => {
  it('renders', () => {
    // Initialize mockstore with empty state
    const initialState = {
      navigation: {
        menuItems: [
          {
            page: '1',
            label: 'Home',
            is_react: false,
            url: './include/home/home.php',
            options: null,
            children: [
              {
                groups: [],
                page: '103',
                label: 'Custom Views',
                is_react: true,
                url: '/home/customViews',
                options: null,
              },
            ],
          },
        ],
      },
    };
    const store = mockStore(initialState);

    const { container } = render(
      <Provider store={store}>
        <Router history={history}>
          <>
          <BreadcrumbWrapper path='/home/customViews' />
          <Switch>
            <Route path="/" render={() => <div>toto</div>} />
          </Switch>
          </>
        </Router>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
