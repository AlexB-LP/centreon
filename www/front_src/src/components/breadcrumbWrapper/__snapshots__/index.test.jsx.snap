// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbWrapper renders 1`] = `
.emotion-0 {
  padding: 0px 16px;
}

.emotion-0 .MuiBreadcrumbs-li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<nav
  aria-label="Breadcrumb"
  class="MuiTypography-root MuiBreadcrumbs-root emotion-0 MuiTypography-body1 MuiTypography-colorTextSecondary"
>
  <ol
    class="MuiBreadcrumbs-ol"
  />
</nav>
`;
